concourse/pipelines/artifact-releaser-test.jsonnet

```diff
@@ -0,0 +1,616 @@
+// Imports.
+local arle = import '../templates/arle.libsonnet';
+local common = import '../templates/common.libsonnet';
+
+// Get the repository given the build.
+local get_repo(build) = if std.startsWith(build, 'deb') then 'guest-arle-autopush-trusty'
+else if std.startsWith(build, 'el') then 'guest-arle-autopush-el7-x86_64'
+else 'guest-arle-autopush';
+
+// Get the universe given the build.
+local get_universe(build) = if std.startsWith(build, 'deb') then 'cloud-apt'
+else if std.startsWith(build, 'el') then 'cloud-yum'
+else 'cloud-yuck';
+
+// Change '-' to '_', mainly used for images
+local underscore(input) = std.strReplace(input, '-', '_');
+
+local upload_arle_autopush_task = {
+  local tl = self,
+
+  package_paths:: error 'must set package_paths in upload_arle_autopush_task',
+  topic:: 'projects/artifact-releaser-autopush/topics/gcs-guest-package-upload-autopush',
+  type:: error 'must set type in upload_arle_autopush_task',
+  universe:: error 'must set universe in upload_arle_autopush_task',
+  repo:: error 'must set repo in upload_arle_autopush_task',
+
+  task: 'upload-arle-autopush-task',
+  config: {
+    platform: 'linux',
+    image_resource: {
+      type: 'registry-image',
+      source: { repository: 'google/cloud-sdk', tag: 'alpine' },
+    },
+    run: {
+      path: 'gcloud',
+      args: [
+        'pubsub',
+        'topics',
+        'publish',
+        tl.topic,
+        '--message',
+        '{"type": "%s", "request": {"gcsfiles": ["%s"], "universe": "%s", "repo": "%s"}}' % [
+          tl.type,
+          tl.package_paths,
+          tl.universe,
+          tl.repo,
+        ],
+      ],
+    },
+  },
+};
+
+
+local upload_arle_autopush_staging = {
+  local tl = self,
+
+  package:: error 'must set package in upload_arle_autopush_staging',
+  gcs_dir:: tl.package + '_',
```

Comment:
why the `_` suffix in the `gcs_dir`? Why not something more meaningful if we really need it?

Reply:
The name is definitely misleading. I think it'd be better to rename it to 'gcs_file_name' instead (the name of the file up until where the package name is defined). The reason for this underscore is because of certain packages (looking at you, _compute-image-windows_) where the package version is not preceded by an underscore like most other packages.